assets/app/scripts/services/data.js

```diff
@@ -639,32 +646,34 @@ DataService.prototype.createStream = function(resource, name, context, opts, isR
     var callback = handle.callback;
     var objectCallback = handle.objectCallback;
     var opts = handle.opts;
+    var key = this._uniqueKeyForResourceContext(resource, objectName, context, _.get(opts, 'http.params'));
+    console.warn('TODO: .watch()/.unwatch() are NOT $http calls, so opts do not have http.params');
```

Comment:
Agree, I prefer the format to stay the same.  If we change it we really would just be coming up with new names for the same idea, which will probably just be confusing.